test/e2e/next-test-lib/next-modes/next-deploy.ts

```diff
@@ -0,0 +1,102 @@
+import path from 'path'
+import execa from 'execa'
+import fs from 'fs-extra'
+import { platform } from 'os'
+import { NextInstance } from './base'
+
+export class NextDeployInstance extends NextInstance {
+  private _cliOutput: string
+  private _buildId: string
+
+  public get buildId() {
+    return this._buildId
+  }
+
+  public async setup() {
+    if (process.env.SITE_URL) {
+      process.env.NEXT_TEST_SKIP_CLEANUP = 'true'
+      console.log('Using SITE_URL', process.env.SITE_URL)
+      this._url = process.env.SITE_URL
+      this._parsedUrl = new URL(this._url)
+      this._buildId = 'build-id'
+      return
+    }
+
+    await super.createTestDir()
+    // We use yarn because it's better at handling local dependencies
+    await execa('yarn', [], {
+      cwd: this.testDir,
+      stdio: 'inherit',
+    })
+    // ensure Netlify CLI is installed
+    try {
+      const res = await execa('ntl', ['--version'])
+      console.log(`Using Netlify CLI version:`, res.stdout)
+    } catch (_) {
+      console.log(`Installing Netlify CLI`)
+      await execa('npm', ['i', '-g', 'netlify-cli@latest'], {
```

Comment:
should we use yarn to install this as we're using yarn above?

Reply:
This one is a global install, so it doesn't really matter